### PR TITLE
UIEH-277: First pass at RAML for providers endpoint

### PR DESCRIPTION
**Pull request by keathleydavidj**
## Purpose
This PR is my first go at documenting mod-kb's `/providers` endpoint requests using RAML 1.0, addressing the ask of [UIEH-277](https://issues.folio.org/browse/UIEH-277). 

## Approach
- Mostly worked through the first RAML tutorial linked below.

#### TODOS and Open Questions
- [ ] Gather and document response shapes/schemas.
- [ ] Pull relevant pieces out into abstractions or partials where it makes sense.

## Learning
[RAML 100 Tutorial](https://raml.org/developers/raml-100-tutorial)
[RAML 200 Tutorial](https://raml.org/developers/raml-200-tutorial)

### CODE REVIEW

File: ramls/eholdings.raml [hunk 1]
```diff
@@ -7,6 +7,109 @@ documentation:
...
+    queryParameters: 
+      q:
+        displayName: Query
+        type: string
+        description: Text entered in search field
+        example: Basket Weaving
+        required: true
+      page:
+        displayName: Page
+        type: number
```
Comment (sduvvuri-ebsco) [inline ramls/eholdings.raml:22]: If there is a way to provide a min value here for page offset, we should specify that as 1, else RM API will fail

File: ramls/eholdings.raml [hunk 2]
```diff
@@ -7,6 +7,109 @@ documentation:
...
+        description: Page offset from which results are retrieved
+        example: 1
+        required: false
+      count:
+        displayName: Count
+        enum: [10, 20, 30]
```
Comment (sduvvuri-ebsco) [inline ramls/eholdings.raml]: similarly, for count, I think we can specify any number between 0 and 100

File: ramls/eholdings.raml [hunk 3]
```diff
@@ -7,6 +7,109 @@ documentation:
...
+        displayName: Sort options
+        enum: [name, relevance]
+        description: Option by which results are sorted
+        default: relevance
+        required: false
+  filterable:
+    queryParameters:
+      filter[selected]:
+        displayName: Selection status
+        enum: [true, false, ebsco]
```
Comment (sduvvuri-ebsco) [inline ramls/eholdings.raml:45]: at least in packages, we support [true, false, ebsco, all]; is that different for providers?

**Comment by sduvvuri-ebsco** [comment]
Overall, this is great @keathleydavidj

File: ramls/eholdings.raml (hunk 3 above)
Comment (keathleydavidj) [reply ramls/eholdings.raml:45]: The way I understood this one when I was playing around with it, selecting `all` just causes this query parameter to be omitted from the request.

File: ramls/eholdings.raml (hunk 1 above)
Comment (keathleydavidj) [reply ramls/eholdings.raml:22]: Reviewing the specification docs revealed that there is indeed a way to restrict the available numbers using `minimum` and `maximum`. -->  https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#number
I have gone ahead and updated with those changes.

**Review by sduvvuri-ebsco — APPROVED** [review]
Looks like we have to update error messages and figure out how to re-use traits for packages. But, I do not want to hold off this PR for that. We shall deal with it in a separate PR. Great job @keathleydavidj

**Pull request merged**